rake_helper.rb

```diff
@@ -0,0 +1,130 @@
+require 'aws-sdk-lambda'
+require 'aws-sdk-cloudwatchevents'
+require 'yaml'
+
+class RakeHelper
+
+  attr_reader(
+    :travis_branch,
+    :aws_access_key_id,
+    :aws_secret_access_key,
+    :aws_configuration,
+    :lambda_client,
+    :yaml,
+    :lambda_config,
+    :function_name,
+    :event
+  )
+
+  def initialize
+    @travis_branch = ENV["TRAVIS_BRANCH"].upcase
+    @aws_access_key_id = ENV["AWS_ACCESS_KEY_ID_#{travis_branch}"]
+    @aws_secret_access_key = ENV["AWS_SECRET_ACCESS_KEY_#{travis_branch}"]
+    @aws_configuration = aws_configuration = {
+      region: 'us-east-1',
+      access_key_id: aws_access_key_id,
+      secret_access_key: aws_secret_access_key
+    }
+    p 'using configuration: ', aws_configuration
+    if configured?
+      @lambda_client = Aws::Lambda::Client.new(aws_configuration)
+      @lambda_config = yaml["deploy"].find {|conf| conf["function_name"].include? travis_branch.downcase}
+    end
+  end
+
+  def configured?
+    access_key_id && secret_access_key
+  end
+
+
+  def update_lambda_configuration
+    unless configured? && lambda_config
+      p 'insufficient configuration'
+      return nil
+    end
+
+
+    vpc_config = lambda_config["vpc"]
+    environment_variables = lambda_config["environment_variables"]&.map {|str| str.split "="}.to_h
+    layers = lambda_config["layers"]
+    updated_lambda_configuration = [
+      :vpc_config,
+      :environment_variables,
+      :layers
+    ]
+      .select {|key| lambda_config[key]}
+      .map {|key| [key, lambda_config[key]]}
+      .to_h
+    p 'updating_function_configuration with: ', updated_lambda_configuration
+    update_configuration_resp = lambda_client.update_function_configuration(updated_lambda_configuration)
+    p 'update_configuration_resp: ', update_configuration_resp
+  end
+
+  def update_event
+    unless lambda_config["events"]
+      p 'no event config'
+      return nil
+    end
+
+    @event = lambda_config["event"]
+    if event.include?("arn")
```

Comment:
When the event type is Kinesis, this needs to take a few extra parameters, in addition to the `arn`, so we can control the lambda invocation:

- `batch_size`
- `maximum_record_age_in_seconds`
- `bisect_batch_on_function_error`
- `maximum_retry_attempts`

I think we should be able to add them as an object in travis and just parse them that way